src/strings/nfa-noodlification.cc

```diff
@@ -50,11 +49,59 @@ void unify_initial_and_final_states(const std::vector<std::shared_ptr<Nfa>>& nfa
 Nfa concatenate_with(const std::vector<std::shared_ptr<Nfa>>& nfas, mata::Symbol delimiter) {
     Nfa concatenation{*nfas[0]};
     for (std::vector<std::shared_ptr<Nfa>>::size_type i = 1; i < nfas.size(); ++i) {
-        concatenation = concatenate_eps(concatenation, *nfas[i], delimiter, true);
+        concatenation = mata::nfa::algorithms::concatenate_eps(concatenation, *nfas[i], delimiter, true);
     }
     return concatenation;
 }
 
+/**
+ * @brief Check if transducer @p nft T is for sure homomorphic, that is whether for each two words u,v, it holds T(u.v) = T(u).T(v).
+ * 
+ * The check is only a heuristic, it can return false even if T is homomorphic. We do this by checking whether there is exactly one
+ * initial and final state (and they are the same), and each (transducer) transition from the initial state does not contain epsilon,
+ * while all other transitions contain epsilon always on the first tape and a some symbol on the second tape.
+ */
+bool is_nft_homomorphic(const std::shared_ptr<Nft> nft) {
```

Comment:
I think this should be a method `nft::Nft::is_homomorphic_heuristic()`.
1. A method is better for is something checks.
2. The suffix `_heuristic` (or `_heur`?) explicitly states that it is a heuristic. Alternatively, we could try to find a more descriptive name for the precise property checked for below. Not sure what that would be, though.

Reply:
I do not want to put it in `nft::Nft`. It is only a heuristic implemented for this specific use case, if we add all these small things to `nft::Nft`, then it will be full of useless methods that nobody can find.

Reply:
Ah, right. Good shout. Marking as resolved.